lua/starfall/libs_sh/builtins.lua

```diff
@@ -1191,6 +1193,30 @@ function builtins_library.enableHud(ply, active)
 	end
 end
 
+--- Restarts a chip owned by yourself.
+-- Only restarts the realm that this gets called in.
+-- @param Entity? chip The chip to restart. If nil, it will restart the current chip.
+function builtins_library.restart(chip)
+	if chip then
+		chip = getent(chip)
+		if not (chip.Starfall and chip.instance) then SF.Throw("Entity has no starfall instance", 2) end
+		if chip.owner ~= instance.player then SF.Throw("You don't own that starfall", 2) end
+	else
+		chip = instance.entity
+	end
+
+	local now = CurTime()
+	if (chip.nextRestartTime or 0) > now then SF.Throw("That starfall is on restart() cooldown", 2) end
+
+	chip.nextRestartTime = now + restartCooldown:GetFloat()
```

Comment:
One last thing, restartCooldown should be clamped with a minimum time. Maybe 0.1 second.